addons/product/product.py

```diff
@@ -685,9 +685,9 @@ def name_search(self, cr, user, name='', args=None, operator='ilike', context=No
             positive_operators = ['=', 'ilike', '=ilike', 'like', '=like']
             ids = []
             if operator in positive_operators:
-                ids = self.search(cr, user, [('default_code','=',name)]+ args, limit=limit, context=context)
+                ids = self.search(cr, user, args + [('default_code','=',name)], limit=limit, context=context)
```

Comment:
AFAIKS ```args``` is supposed to be a generic list of tuples in the form accepted by the search method. Adding a tuple before args, like ```[('default_code','=',name)]+ args```, means adding that condition with AND.
So, ```args + [('default_code','=',name)]``` could produce a not valid list of tuples.
I think you should build a correct ```args``` in your module